### PR TITLE
Improve color palette doc

### PR DESCRIPTION
**Pull request by couet**
as requested here:

https://github.com/root-project/root/issues/8098

### CODE REVIEW

File: core/base/src/TColor.cxx [hunk 1]
```diff
@@ -326,14 +327,36 @@ kWaterMelon=108,      kCool=109,        kCopper=110,
...
 ~~~
 
+One can immediately see the [disadvantages of the Rainbow color map](https://root.cern.ch/rainbow-color-map),
+which is misleading for colour-blinded people in a 2D plot (not so much in a 3D surfaces).
+
+The `kCMYK` palette, for instance, is also not great because it's dark, then lighter, then
+half-dark again. Some others, like `kAquamarine`, have almost no contrast therefore it would
+be almost impossible (for a color blind person) to see something with a such palette.
+
+Therefore the palettes are classified in two categories: those which are  Colour Vision Deficiency
```
Comment (ferdymercury) [inline core/base/src/TColor.cxx:337]: two spaces (are  Colour)

File: core/base/src/TColor.cxx [hunk 2]
```diff
@@ -326,14 +327,36 @@ kWaterMelon=108,      kCool=109,        kCopper=110,
...
+friendly and those which are not.
+
+
+An easy way to classify the palettes is to turn them into grayscale using TCanvas::SetGrayscale().
+The grayscale  version of a palette should be as proportional as possible, and monotonously
```
Comment (ferdymercury) [inline core/base/src/TColor.cxx:342]: (two spaces)

File: documentation/doxygen/makeinput.sh [hunk 3]
```diff
@@ -9,69 +9,69 @@
 echo "INPUT = ./mainpage.md                    \\" > Doxyfile_INPUT
 
 echo "        ../../core/base/                 \\" >> Doxyfile_INPUT
-echo "        ../../core/dictgen/              \\" >> Doxyfile_INPUT
```
Comment (ferdymercury) [inline documentation/doxygen/makeinput.sh:12]: This slipped I think

**Comment by couet** [comment]
I added the link to the Nature article.

File: core/base/src/TColor.cxx [hunk 4]
```diff
@@ -64,6 +64,8 @@ The color creation and management class.
   - [Gray scale view of of canvas with colors](\ref C04)
   - [Color palettes](\ref C05)
   - [High quality predefined palettes](\ref C06)
+    - [Colour Vision Deficiency (CVD) friendly palettes](\ref C06a)
+    - [Non Colour Vision Deficiency (CVD) friendly palettes](\ref C06b)
```
Comment (bellenot) [inline core/base/src/TColor.cxx:68]: Is the change of indentation intentional?

**Review by bellenot — APPROVED** [review]
LGTM, beside the indentation question

File: core/base/src/TColor.cxx (hunk 4 above)
Comment (couet) [reply core/base/src/TColor.cxx:68]: Yes they are subsections of the one above

**Pull request merged**